File: tfidf/bin/data/to_lbl.rb
#!/home/takayuk/local/bin/ruby -Ku
# encoding: utf-8
# coding: utf-8

url=open(ARGV[0]).readlines.map{|v|v.chomp}
doc=open(ARGV[1]).readlines.map{|v|v.chomp}.map{|w|w.split(" ")}

open(ARGV[2],"w"){|file|
  file.puts "#{doc.size} #{url.size-1}"
  doc.each_with_index{|d, i|
    wlist=d.map{|v|
      url.index(v.gsub(/[^a-zA-Z0-9]/,"").downcase)
    }

    wdic=Hash.new(0)
    wlist.reject{|v|v=="" or v.nil?}.each{|w|wdic[w]+=1}
    buf=""
    wdic.each{|k,v| 
      p k if k.nil?
      buf+=" #{k}:#{v}"
    }
    file.puts "#{i+1} #{wdic.size}#{buf}"
  }
}

File: tfidf/bin/data/to_lbl.py
#!/home/takayuk/local/bin/python
# encoding: utf-8
# coding: utf-8

import sys
sys.path.append(".")

import re
import tf_idf as tfidf

args=sys.argv

def to_wordurl(docs):
    words_dic={}
    tfdic=tf_idf.tf(docs)
    
    return tfdic.keys()

with file(args[1]) as url_file:
    urls=[url.strip("\n") for url in url_file.readlines()]

with file(args[2]) as text:
    docs=[doc.split(" ") for doc in [line.rstrip("\n") for line in text.readlines()]]

with file(args[3],"w") as out_file:



    #docs=[[re.sub("[^a-zA-Z]", "", word) for word in doc] for doc in docs]

#tfdic=tfidf.tf(docs)
#dfdic=tfidf.df(tfdic)

"""
wordurl=to_wordurl(docs)
try:
  for i in xrange(wordurl.count("")):
    wordurl.remove("")
except ValueError:
  print(wordurl)

with file(args[2],"w") as url:
  url.write(str(len(wordurl))+"\n")
  for word in wordurl:
    url.write(word+"\n")
"""

